packages/labs/analyzer/src/lib/javascript/modules.ts

```diff
@@ -40,29 +53,27 @@ export const getModule = (
     // separators; since sourcePath uses OS separators, normalize
     // this so that all our model paths are OS-native
     jsPath: absoluteToPackage(
-      path.normalize(jsPath) as AbsolutePath,
-      programContext.packageRoot as AbsolutePath
+      analyzer.path.normalize(jsPath) as AbsolutePath,
+      rootDir
     ),
     sourceFile,
+    packageJson,
```

Comment:
Modules need to know their package name in order to create references for things in them.

For package-local references (references to things within a given module or relative imports from it), as opposed to references imported from separate packages), we need to get the package name and jsPath for the given module. That logic is in `reference.ts`.

In the previous factoring, the `ProgramContext` held the `packageJson` for the package being analyzed. Since we want to be able to get a module standalone without analyzing an entire package (for plugin usage), both `getModule` and `getPackage` look up the package.json (although `getPackage` is factored to pass the package info into `getModule` so that we don't look it up over and over while analyzing an entire package.

There are TODOs in the code for figuring out a caching strategy for package.json, but haven't tackled that yet.